### PR TITLE
Do not run PBM until shard is not assembled

### PR DESCRIPTION
**Pull request by blez**

### CODE REVIEW

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| mongodb 3.6 | failed |
| mongodb 4.0 | failed |
| mongodb 4.2 | failed |

**Comment by blez** [comment]
BTW, this
```
exit_status=$?
```
always equal to 0. Even if error occurred.

File: docker/start-agent.sh [hunk 1]
```diff
@@ -6,16 +6,33 @@ set -o xtrace
...
+			echo "waiting for sharded scluster"
+			mongo "${PBM_MONGODB_URI}" --eval="db.isMaster().\$configServerState.opTime.ts" --quiet | tee "$OUT"
+			exit_status=$?
+			mongo "${PBM_MONGODB_URI}" --eval="db.isMaster().configsvr" --quiet | tail -n 1 | tee "$OUT_CFG"
+			exit_status_cfg=$?
+
+			ts=$(grep -E '^Timestamp\([0-9]+, [0-9]+\)$' "$OUT")
+			isCfg=$(grep -E '^2$' "$OUT_CFG")
+
+			if [[ "${exit_status}" == 0 && "${ts}" ]] || [[ "${exit_status_cfg}" == 0 && "${isCfg}" ]]; then
```
Comment (delgod) [inline docker/start-agent.sh:22]: can we put into comment why we check one or another command?

**Review by delgod — CHANGES_REQUESTED** [review]

File: docker/start-agent.sh (hunk 1 above)
Comment (blez) [reply docker/start-agent.sh:26]: https://github.com/percona/percona-backup-mongodb/pull/459/commits/8f343ef4ddef35decc99f3316dfa0a9b745895e4
In short  - the commands do same check as PBM did here
https://github.com/percona/percona-backup-mongodb/blob/main/pbm/bsontypes.go#L49-L51

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| mongodb 3.6 | failed |
| mongodb 4.0 | failed |
| mongodb 4.2 | failed |

**Review by dAdAbird — APPROVED** [review]

**Review by delgod — APPROVED** [review]

**Pull request merged**